calico_node/startup/startup.go

```diff
@@ -539,6 +542,37 @@ func configureASNumber(node *api.Node) {
 	}
 }
 
+// getIPv6Pool return a random generated ULA IPv6 prefix generated following rfc4193#section-3.2.2
+// The Pool is generated with a concatenation of Unix timestamps + fe80:: base IPv6 hased with SHA-1
+func getIPv6Pool() string {
+	adm := os.Getenv("IP_AUTODETECTION_METHOD")
+	cidr := autoDetectCIDR(adm, 6)
+	eui := cidr.String()
+	if eui != "" {
+		date := fmt.Sprint(time.Now().Unix())
+		d := []byte(date)
+		h := sha1.New()
+		h.Write(d)
+		h.Write([]byte(strings.Replace(eui, ":", "", -1)))
+		sum := hex.EncodeToString(h.Sum(nil))
+		// IPv6 Random Pool generation
+		buf := &bytes.Buffer{}
```

Comment:
Why not work with the bytes directly instead of using a string and converting everything:
```
randomBytes, err := rand.Read(5) // 40 bits
if err != nil { /* panic */)
addr := make(net.IP, 16)
addr[0] = 0xfd
copy(addr[1:6], randomBytes)
final := addr.String() + "/64"
```